Add Chainsaw weapon to Hunter's Glee?

Boomertaters opened the pull request.

Boomertaters committed:
Added the chainsaw weapon

Boomertaters committed:
Added chainsaw materials

Boomertaters committed:
Added chainsaw model

Boomertaters committed:
Added chainsaw sounds + AR3 sounds

Boomertaters committed:
Added chainsaw hand material

I don't know why this is needed, but the chainsaw needs this in order to not show as missing textures.

Boomertaters committed:
Added chainsaw spawnmenu icon

Boomertaters committed:
Add new AR3 sound effects

Boomertaters committed:
Implement chainsaw purchase and adjust weights

StrawWagen committed:
style

StrawWagen committed:
style 2

Boomertaters committed:
Do what straw said to do

.

Boomertaters committed:
Fix the slient ar3 shooting sounds

StrawWagen commented on gamemodes/hunters_glee/gamemode/sh_shopitems.lua:
```diff
@@ -3471,10 +3480,25 @@ local defaultItems = {
+        name = "Chainsaw",
+        desc = "A roaring beast of blood and metal, nothing survives the bite of its spinning teeth... \nMetal’s the only thing that bites back...",
+        cost = 0,
+        skullCost = 3,
```
too cheap, 4 skulls or 5 skulls prob

StrawWagen commented on lua/weapons/termhunt_ar3.lua:
```diff
@@ -30,7 +30,7 @@ SWEP.WorldModel = "models/weapons/tfa_mmod/w_ar3.mdl"
 SWEP.UseHands = true
 
-SWEP.ShootSound = "Weapon_FuncTank.Single"
+SWEP.ShootSound = ( "ar3/ar3_fire" .. math.random(1,3) .. ".wav" )
```
this only picks one of the sounds for the whole session

StrawWagen commented on lua/weapons/termhunt_ar3.lua:
```diff
@@ -189,7 +190,7 @@ function SWEP:Reload()
         if not IsValid( self ) then return end
 
-        self:EmitSound( "weapons/shotgun/shotgun_cock.wav" )
+        self:EmitSound( "ar3/ar3_pump.wav" )
```
this sounds better 👍

StrawWagen reviewed (changes requested):
I didn't leave any specific comments on the chainsaw file, because all the problems are too big to be able to focus on.

For the looping sounds, you're starting/stopping them in a bunch of different spots
You should instead create and manage the sound in one place, inside Think.
Then dynamically restart it with a different soundpath depending on whether the saw is... idle, ( kill the sound ) revved up, ( restart the sound, with a new soundpath ) or attacking, ( same thing )
The best way to do this will be to have some logic that picks the soundpath depending on the new NetworkVars, then some logic that manages the sound, killing/restarting it if the path changes.
This logic should probably be on the server only? I think it might work fine shared though.

Note how i made 2 sound managers [here in the tau cannon.](https://github.com/StrawWagen/hunters_glee/blob/main/lua/weapons/termhunt_taucannon/shared.lua#L229-L294)
And note [how many places they're killed and cleaned up with the KillSound function](https://github.com/StrawWagen/hunters_glee/blob/main/lua/weapons/termhunt_taucannon/shared.lua#L192-L201)

Boomertaters commented:
ok now i wake up now from my gleeumber

Boomertaters committed:
Do what straw said to do PART 2

Boomertaters committed:
four

Boomertaters committed:
fix sound

Boomertaters committed:
Fix deploy sound for ar3

Boomertaters commented:
Oh i forgot to fix the world model 
w/e it's nothing to worry about

Boomertaters committed:
Forgot to remove this

Boomertaters commented:
keep forgetting that i don't need number thingys anymore, sorry bout that straw
so used to have them that its just a force of habit to add them

Boomertaters committed:
fix annoying click sound playing whenever pressing right click

Boomertaters committed:
Added secondary things back

I thought i didn't need the secondary stuff because i added the thing that turns secondaryattack off, my fault

Boomertaters commented:
So many mistakes

Boomertaters committed:
add new instruction

Boomertaters commented:
ok i'm just closing this

Boomertaters commented:
i'll make a new one
